graspologic/pipeline/embed/laplacian_spectral_embedding.py

```diff
@@ -0,0 +1,247 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+import numbers
+import warnings
+from typing import Optional, Union
+
+import networkx as nx
+import numpy as np
+
+from graspologic.embed import LaplacianSpectralEmbed
+from graspologic.preconditions import (
+    check_argument,
+    check_argument_types,
+    check_optional_argument_types,
+    is_real_weighted,
+)
+from graspologic.utils import is_fully_connected, pass_to_ranks, remove_loops
+
+from . import __SVD_SOLVER_TYPES  # from the module init
+from ._elbow import _index_of_elbow
+from .embeddings import Embeddings
+
+__FORMS = ["DAD", "I-DAD", "R-DAD"]
+
+
+def laplacian_spectral_embedding(
+    graph: Union[nx.Graph, nx.DiGraph],
```

Comment:
you also allow Ordered version of these classes

Reply:
@daxpryce / @jonmclean  just a question for the future - how does typing/beartype handle subtypes of the type that you specify in the decorator? asking cause in this case I believe OrderedGraph is a subtype of Graph https://networkx.org/documentation/stable/_modules/networkx/classes/ordered.html#OrderedGraph and I'm just wondering what the convention should be going forward?